### PR TITLE
Test: SynapseRouterV2 Integration

### PR DESCRIPTION
**Pull request by mikeyrf**
## Description

Integration tests on a few chains for `SynapseRouterV2.sol`.

## Checklist

- [ ] New Contracts have been tested
- [ ] Lint has been run
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by mikeyrf** [comment]
@ChiTimesChi core of Router V2 integration tests is finished, so could use an initial review. The Arbitrum implementation only considers expected tokens for nUSD related pool (USD stablecoin bridging) with CCTP and Synapse bridge modules. Can add more bridge tokens if you like to this as well.

**Comment by mikeyrf** [comment]
@ChiTimesChi refactor finished and ready for review for you.

File: test/utils/IntegrationUtils.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    }
+
+    function afterBlockchainForked() public virtual {}
+
+    function getChainId() public view returns (uint256 id) {
+        assembly {
+            id := chainid()
+        }
+    }
```
Comment (ChiTimesChi) [inline test/utils/IntegrationUtils.sol:36]: With pragma 0.8.17 `block.chainid` could be used instead of wrapping this into a function

**Review by ChiTimesChi — CHANGES_REQUESTED** [review]
Tests look great, nice work! Requesting some final changes, then it should be good to go for the Arbitrum tests :rocket:

File: test/router/v2/integration/SynapseRouterV2.Integration.Arbitrum.t.sol [hunk 2]
```diff
@@ -0,0 +1,820 @@
...
+        assertEq(query.minAmountOut, 0);
+        assertEq(query.deadline, 0);
+        assertEq(query.rawParams, bytes(""));
+    }
+
+    /// @dev CCTP has max bridged amount out of 1M USDC
+    function testGetOriginAmountOut_inUSDCe_outUSDC() public {
+        address tokenIn = USDC_E;
+        string[] memory tokenSymbols = new string[](1);
+        tokenSymbols[0] = "CCTP.USD";
```
[the remark this answers is not in the document]
Comment (mikeyrf) [reply test/router/v2/integration/SynapseRouterV2.Integration.Arbitrum.t.sol:253]: Good catch ty!

**Comment by mikeyrf** [comment]
@ChiTimesChi Ready for final go ahead then can merge 🖖

**Review by ChiTimesChi — APPROVED** [review]
Very solid job! LGTM

**Pull request merged**